### PR TITLE
Added feature gating

### PR DESCRIPTION
**Pull request by clintonb**
@dsjen @rocha

### CODE REVIEW

File: analytics_dashboard/courses/tests/test_views.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 import analyticsclient.activity_type as AT
 from analyticsclient.exceptions import NotFoundError
+from waffle import Switch
```
Comment (dsjen) [inline analytics_dashboard/courses/tests/test_views.py]: I believe we're ordering the imports with 3rd party packages up top.

**Comment by rocha** [comment]
I didn't know about waffle. It looks pretty neat! The remainder task is to put each of the components under waffle features, both for html rendering and the data requests.

**Comment by clintonb** [comment]
I have updated the UI to use feature gates to control the display of items on the navigation bar.

File: analytics_dashboard/analytics_dashboard/context_processors.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+def _is_navbar_item_enabled(item):
+    switch_name = item.get('switch', None)
+    if switch_name:
+        return switch_is_active(switch_name)
+    return True
+
+
+def navbar(request):
+    course_id = request.course_id
```
Comment (rocha) [inline analytics_dashboard/analytics_dashboard/context_processors.py:15]: Can you add a little comment of what is the purpose of this processor. It is easy to get it from the context of the pull request, but someone just reading the code may not know why is this here.

File: analytics_dashboard/courses/templates/courses/engagement.html [hunk 3]
```diff
@@ -10,245 +11,252 @@
...
-            Download a CSV file of this Data
+          <i class="ico fa fa-download"></i>
+          Download a CSV file of this Data
         </a>
+      </div>
     </div>
-</section>
+  {% endswitch %}
+  </section>
 {% endblock content %}
```
Comment (rocha) [inline analytics_dashboard/courses/templates/courses/engagement.html:262]: Making indentation/formatting changes is ok, but I prefer if they are in separate PRs. They make the diffs hard to read.

**Comment by rocha** [comment]
:rocket: 

Please add a comment to the `navbar` context processor before merging.
